historical_system_profiles/archiver.py

```diff
@@ -101,11 +102,18 @@ def _archive_profile(data, ptc, logger, notification_service):
         )
         _record_duplicate_message(host, request_id, ptc)
     else:
+        # remove keys we do not care about
+        if groups:
+            groups = [
+                {key: group[key] for key in ["id", "name"] & group.keys()} for group in groups
```

Comment:
Can we extract the expected keys into a constant?
Can this be moved to a function?

Reply:
I hope solved!